container-core/src/main/java/com/yahoo/restapi/RestApi.java

```diff
@@ -0,0 +1,367 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.restapi;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.yahoo.container.jdisc.HttpRequest;
+import com.yahoo.container.jdisc.HttpResponse;
+import com.yahoo.jdisc.http.HttpRequest.Method;
+import com.yahoo.slime.Slime;
+import com.yahoo.slime.SlimeUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalLong;
+
+/**
+ * Rest API routing and response serialization
+ *
+ * @author bjorncs
+ */
+public class RestApi {
+
+    private final Route defaultRoute;
+    private final List<Route> routes;
+    private final List<ExceptionMapperHolder<?>> exceptionMappers;
+    private final List<ResponseMapperHolder<?>> responseMappers;
+    private final ObjectMapper jacksonJsonMapper;
+
+    private RestApi(Builder builder) {
+        ObjectMapper jacksonJsonMapper = builder.jacksonJsonMapper != null ? builder.jacksonJsonMapper : JacksonJsonMapper.instance;
+        this.defaultRoute = builder.defaultRoute != null ? builder.defaultRoute : createDefaultRoute();
+        this.routes = List.copyOf(builder.routes);
+        this.exceptionMappers = combineWithDefaultExceptionMappers(builder.exceptionMappers);
+        this.responseMappers = combineWithDefaultResponseMappers(builder.responseMappers, jacksonJsonMapper);
```

Comment:
combineWithDefaultExceptionMappers() and combineWithDefaultResponseMappers() should probably be part of the builder.  It would allow a client of the builder to override/clear the defaults, if we want that later.  It would allow the builder to document the defaults, and that adding mappers will be in addition to the defaults.

Reply:
Overriding is already possible as the default mappers acts as fallbacks, and a user can override a mappers behaviour just by specifying a custom mapper for the same type. Clearing all mappers is not - that could be solved by adding `disableBuiltinExceptionMappers`/`disableBuiltinResponseMappers` to builder.